testsuite/forge/src/main.rs

```diff
@@ -310,22 +325,74 @@ impl Test for EmitTransaction {
 impl NetworkTest for EmitTransaction {
     fn run<'t>(&self, ctx: &mut NetworkContext<'t>) -> Result<()> {
         let duration = Duration::from_secs(10);
-        let rng = SeedableRng::from_rng(ctx.core().rng()).unwrap();
-        let validator_clients = ctx
-            .swarm()
-            .validators()
-            .into_iter()
-            .map(|n| n.async_json_rpc_client())
-            .collect_vec();
-        let mut emitter = TxnEmitter::new(ctx.swarm().chain_info(), rng);
-        let rt = Runtime::new().unwrap();
-        let stats = rt
-            .block_on(emitter.emit_txn_for(duration, EmitJobRequest::default(validator_clients)))
-            .unwrap();
+        let stats = generate_traffic(ctx, duration).unwrap();
         ctx.report
             .report_txn_stats(self.name().to_string(), stats, duration);
         ctx.report.print_report();
 
         Ok(())
     }
 }
+
+fn batch_update<'t>(
+    validators_to_update: Vec<String>,
+    image_tag: &str,
+    helm_repo: &str
+) -> Result<()> {
+    for validator in validators_to_update {
+        let _ = set_validator_image_tag(
+            &validator,
+            image_tag,
+            helm_repo,
+        );
+    }
+
+    Ok(())
+}
+
+#[derive(Debug)]
+struct CompatibilityTest;
+
+impl Test for CompatibilityTest {
+    fn name(&self) -> &'static str {
+        "compatibility_test"
+    }
+}
+
+impl NetworkTest for CompatibilityTest {
```

Comment:
I already have a compatibility test defined in https://github.com/diem/diem/pull/8872. Is there a reason you're creating a new one instead of reusing that code? If the answer is that it hasn't been merged yet, then the easiest solution is to just temporarily build on top of that PR until its merged.